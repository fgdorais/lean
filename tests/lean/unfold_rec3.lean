open nat

definition nrec [recursor] := @nat.rec

definition myadd x y :=
nrec y (λ n r, succ r) x

theorem myadd_zero : ∀ n, myadd n 0 = n :=
begin
  intro n, induction n with n ih,
  reflexivity,
  esimp [myadd],
  state,
  rewrite ih
end
