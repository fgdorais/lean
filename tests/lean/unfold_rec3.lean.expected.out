unfold_rec3.lean:13:2: proof state
n : ℕ,
ih : myadd n 0 = n
⊢ succ (myadd n 0) = succ n
